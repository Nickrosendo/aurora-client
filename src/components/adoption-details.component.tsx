import React from 'react';
import {
  Container,
  Box,
  Icon,
  Img,
  Badge,
  Text,
  Flex,
  Grid,
  GridItem,
  Button,
  useStyleConfig,
} from '@chakra-ui/react';
import { MdLocationOn } from 'react-icons/md';
import {
  FaUsers,
  FaWhatsapp,
  FaInstagram,
  FaEnvelope,
  FaPlus,
} from 'react-icons/fa';

import { mock_data } from '@root/data';
import { getRandomColor } from '@root/helpers';

export interface AdoptionDetailsProps {
  adoptionId: string;
}

export const AdoptionDetails: React.FC<AdoptionDetailsProps> = ({
  adoptionId = '',
}) => {
  const styles = useStyleConfig('AdoptionListItem');
  const item = mock_data[0];

  return (
    <Container maxW="xl" w="full" overflow="hidden" title="adoption-details">
      <Box
        w="full"
        p="2"
        borderWidth="1px"
        __css={styles}
        borderRadius="lg"
        overflow="hidden"
        title="adoption-list-item"
        cursor="pointer"
      >
        <Box h="30rem">
          <Img
            src={item.imageUrl}
            alt={item.title}
            boxSize="100%"
            borderRadius="lg"
            objectFit="cover"
          />
        </Box>

        <Box p="6">
          <Grid
            templateRows="repeat(4, 1fr)"
            templateColumns="repeat(2, 50%)"
            gap={4}
          >
            <GridItem rowSpan={1} colSpan={1}>
              {item.owner && (
                <Flex gap={2} alignItems="center" my="2">
                  <Img
                    borderRadius="full"
                    src={item.owner?.image}
                    alt={item.owner?.name}
                    boxSize="2rem"
                  />
                  <Text fontWeight="semibold"> {item.owner.name} </Text>
                </Flex>
              )}
            </GridItem>
            <GridItem rowSpan={1} colSpan={1}>
              <Flex
                gap={1}
                alignItems="center"
                w="full"
                justifyContent="flex-end"
              >
                <Icon as={FaWhatsapp} h={'2rem'} w={'2rem'} />
                <Icon as={FaInstagram} h={'2rem'} w={'2rem'} />
                <Icon as={FaEnvelope} h={'2rem'} w={'2rem'} />
              </Flex>
            </GridItem>
            <GridItem rowSpan={1} colSpan={2}>
              <Box gap="2" display="flex" alignItems="baseline">
                {item.tags &&
                  item.tags.map((tag: string) => (
                    <Badge
                      key={tag}
                      id={tag}
                      borderRadius="full"
                      px="2"
                      colorScheme={getRandomColor()}
                    >
                      {tag}
                    </Badge>
                  ))}
              </Box>
            </GridItem>
            <GridItem rowSpan={3} colSpan={2}>
              <Box mt="1" fontWeight="bold" as="h4" lineHeight="tight">
                {item.title}
              </Box>
              <Box
                mt="1"
                minHeight="3rem"
                fontWeight="normal"
                borderRadius="md"
                color="white"
                bg="brand.500"
                as="p"
                lineHeight="tight"
                p="2"
              >
                Cachorro fofinho que precisa de um lar.
              </Box>
            </GridItem>

            <GridItem rowSpan={1} colSpan={1}>
              <Flex>
                <Text mr="1"> age: </Text>
                {item.age}
                <Box as="span" color="gray.600" fontSize="sm">
                  / {item.ageType}
                </Box>
              </Flex>
            </GridItem>

            <GridItem rowSpan={1} colSpan={1}>
              <Flex gap={1} alignItems="center">
                <Text mr="1"> location: </Text>
                <Icon as={MdLocationOn} h={'1rem'} w={'1rem'} />
                {item.location}
              </Flex>
            </GridItem>

            <GridItem rowSpan={1} colSpan={2}>
              <Flex gap={1} alignItems="center" fontSize="1.5rem">
                <Icon as={FaUsers} h={'2rem'} w={'2rem'} />
                {item.pretenders}
                <Button ml="2">
                  <Icon as={FaPlus} h={'2rem'} w={'2rem'} />
                </Button>
              </Flex>
            </GridItem>
          </Grid>
        </Box>
      </Box>
    </Container>
  );
};
