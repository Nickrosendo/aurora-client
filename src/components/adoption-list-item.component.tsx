import React from 'react';
import {
  Box,
  Icon,
  Img,
  Badge,
  Text,
  Flex,
  useStyleConfig,
} from '@chakra-ui/react';
import { MdLocationOn } from 'react-icons/md';
import { FaUsers } from 'react-icons/fa';

import { IAdoptionItem } from '@root/types';

export interface AdoptionListItemProps {
  item: IAdoptionItem;
}

export const AdoptionListItem: React.FC<AdoptionListItemProps> = ({
  item = { title: 'foo' },
}) => {
  const styles = useStyleConfig('AdoptionListItem');
  const getRandomColor = () => {
    const colors = ['orange', 'purple', 'teal', 'red', 'green'];
    const randomIndex = Math.round(Math.random() * colors.length);
    return colors[randomIndex];
  };

  const property = {
    imageUrl: 'https://bit.ly/2Z4KKcF',
    imageAlt: 'Rear view of modern home with pool',
    beds: 3,
    baths: 2,
    title: 'Modern home in city center in the heart of historic Los Angeles',
    formattedPrice: '$1,900.00',
    reviewCount: 34,
    rating: 4,
  };

  return (
    <Box
      maxW="lg"
      w="full"
      p="2"
      borderWidth="1px"
      __css={styles}
      borderRadius="lg"
      overflow="hidden"
      title="adoption-list-item"
    >
      <Box h="30rem">
        <Img
          src={item.imageUrl}
          alt={item.title}
          boxSize="100%"
          borderRadius="lg"
          objectFit="cover"
        />
      </Box>

      <Box p="6">
        {item.owner && (
          <Flex gap={2} alignItems="center" my="2">
            <Img
              borderRadius="full"
              src={item.owner?.image}
              alt={item.owner?.name}
              boxSize="2rem"
            />
            <Text fontWeight="semibold"> {item.owner.name} </Text>
          </Flex>
        )}

        <Box gap="2" display="flex" alignItems="baseline">
          {item.tags &&
            item.tags.map((tag: string) => (
              <Badge
                key={tag}
                id={tag}
                borderRadius="full"
                px="2"
                colorScheme={getRandomColor()}
              >
                {tag}
              </Badge>
            ))}
        </Box>

        <Box mt="1" fontWeight="bold" as="h4" lineHeight="tight">
          {item.title}
        </Box>

        <Box>
          {item.age}
          <Box as="span" color="gray.600" fontSize="sm">
            / {item.ageType}
          </Box>
        </Box>

        <Flex gap={1} alignItems="center">
          <Icon as={MdLocationOn} h={'1rem'} w={'1rem'} />
          {item.location}
        </Flex>

        <Flex gap={1} alignItems="center">
          <Icon as={FaUsers} h={'1rem'} w={'1rem'} />
          {item.pretenders}
        </Flex>
      </Box>
    </Box>
  );
};
