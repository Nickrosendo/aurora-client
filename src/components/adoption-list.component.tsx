import React from "react";
import { Box, VStack } from "@chakra-ui/react";

import { AdoptionListItem } from "./adoption-list-item.component";
import type { IAdoptionItem } from "@root/types";

export interface AdoptionListProps {
  items: Array<IAdoptionItem>;
}

export const AdoptionList: React.FC<AdoptionListProps> = ({ items = [] }) => {
  return (
    <VStack spacing="5" title="adoption-list-container" w="full" p="2">
      {items?.map((item: IAdoptionItem) => (
        <AdoptionListItem item={item} key={item.title} />
      ))}
    </VStack>
  );
};
