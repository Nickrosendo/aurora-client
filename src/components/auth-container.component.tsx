import React from 'react';
import { Box } from '@chakra-ui/react';

export const AuthContainer: React.FC = () => {
  return <Box title="auth-container"> AuthContainer </Box>;
};
