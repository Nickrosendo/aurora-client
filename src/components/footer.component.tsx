import React from 'react';
import { Box } from '@chakra-ui/react';

export const Footer: React.FC = () => {
  return <Box title="footer"> footer </Box>;
};
