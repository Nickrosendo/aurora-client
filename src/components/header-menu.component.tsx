import React from 'react';
import {
  Icon,
  IconButton,
  Drawer,
  DrawerOverlay,
  DrawerContent,
  DrawerBody,
  DrawerCloseButton,
  useDisclosure,
} from '@chakra-ui/react';
import { MdMenu } from 'react-icons/md';

export const HeaderMenu: React.FC = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  const iconBtnRef = React.useRef<HTMLButtonElement>(null);

  return (
    <>
      <IconButton
        aria-label="open menu"
        title="open-drawer-menu"
        role="button"
        ref={iconBtnRef}
        icon={<Icon as={MdMenu} h={'1.5rem'} w={'1.5rem'} />}
        onClick={onOpen}
      />
      <Drawer
        isOpen={isOpen}
        placement="left"
        onClose={onClose}
        finalFocusRef={iconBtnRef}
      >
        <DrawerOverlay />
        <DrawerContent>
          <DrawerCloseButton />
          <DrawerBody>foo</DrawerBody>
        </DrawerContent>
      </Drawer>
    </>
  );
};
