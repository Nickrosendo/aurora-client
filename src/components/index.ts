export * from './adoption-list.component';
export * from './adoption-list-item.component';
