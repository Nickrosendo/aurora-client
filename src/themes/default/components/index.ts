export * from './input.theme';
export * from './adoption-list-item.theme';
