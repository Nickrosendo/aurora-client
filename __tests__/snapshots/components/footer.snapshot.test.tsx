import React from 'react';
import renderer from 'react-test-renderer';
import { Footer } from '@root/components';

it('should render Footer component matching snapshot', () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
