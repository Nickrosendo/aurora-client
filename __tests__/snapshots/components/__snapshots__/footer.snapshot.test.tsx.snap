// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component matching snapshot 1`] = `
<div
  className="css-0"
  title="footer"
>
   footer 
</div>
`;

exports[`should render SignIn component matching snapshot 1`] = `
<div
  className="css-0"
  title="sign-in"
>
   signIn 
</div>
`;
